src/config/logger.ts

```diff
@@ -101,12 +172,49 @@ const createLoggerOptions = (label: string): LoggerOptions => {
 }
 
 /**
- * Create a new winston logger with given label
- * @param label The label of the logger
+ * Returns a label from the given module
+ * @example loaders/index.ts
+ * @param callingModule
+ */
+const getModuleLabel = (callingModule: NodeModule) => {
+  // Remove the file extension from the filename and split with path separator.
+  const parts = callingModule.filename.replace(/\.[^/.]+$/, '').split(path.sep)
+  // Join the last two parts of the file path together.
+  return path.join(parts[parts.length - 2], parts.pop())
+}
+
+/**
+ * Overrides the given winston logger with a new signature, so as to enforce a
+ * log format.
+ * @param logger the logger to override
+ */
+const createCustomLogger = (logger: Logger) => {
+  return {
+    info: ({ message, meta }: Omit<CustomLoggerParams, 'error'>) =>
+      logger.info(message, { meta }),
+    warn: ({ message, meta, error }: CustomLoggerParams) => {
+      if (error) {
+        return logger.warn(message, { meta }, error)
```

Comment:
why do you need to keep destructuring and restructuring?

Reply:
Forces correct logs, if we pass in error even if its undefined, the key will still appear in the logging output:

```
{
  error: undefined,
  ...rest
}
```

Reply:
I mean for `meta`, why do you destructure it in the function signature then pass it as `{ meta }`?

Reply:
Ah, I'll need to add a docs for that. `info.meta` is actually a real subkey in winston, meaning everything will be destructured if we leave it in like that. Encapsulating it as an object ensures that the JSON will be of the correct shape.

https://github.com/winstonjs/winston#streams-objectmode-and-info-objects